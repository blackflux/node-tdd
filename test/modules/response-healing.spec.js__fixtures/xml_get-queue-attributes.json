{
  "fn": "call",
  "params": [
    "sqs:GetQueueAttributesCommand",
    {
      "QueueUrl": "some-queue-url",
      "AttributeNames": [
        "ApproximateNumberOfMessages",
        "ApproximateNumberOfMessagesDelayed",
        "ApproximateNumberOfMessagesNotVisible"
      ]
    }
  ],
  "cassette": [
    {
      "scope": "https://sqs.us-west-2.amazonaws.com:443",
      "method": "POST",
      "path": "/",
      "body": "QueueUrl=https%3A%2F%2Fsqs.us-west-2.amazonaws.com%2F012345678910%2Fsome-queue-url&AttributeName.1=ApproximateNumberOfMessages&AttributeName.2=ApproximateNumberOfMessagesDelayed&AttributeName.3=ApproximateNumberOfMessagesNotVisible&AttributeName.4=QueueArn&Action=GetQueueAttributes&Version=2012-11-05",
      "status": 200,
      "response": "<?xml version=\"1.0\"?><GetQueueAttributesResponse xmlns=\"http://queue.amazonaws.com/doc/2012-11-05/\"><GetQueueAttributesResult><Attribute><Name>ApproximateNumberOfMessages</Name><Value>1</Value></Attribute><Attribute><Name>ApproximateNumberOfMessagesDelayed</Name><Value>0</Value></Attribute><Attribute><Name>ApproximateNumberOfMessagesNotVisible</Name><Value>0</Value></Attribute><Attribute><Name>QueueArn</Name><Value>arn:aws:sqs:us-west-2:012345678910:some-queue-arn</Value></Attribute></GetQueueAttributesResult><ResponseMetadata><RequestId>b8ca4c86-2fd1-5481-9e72-160c0ef00529</RequestId></ResponseMetadata></GetQueueAttributesResponse>",
      "responseIsBinary": false
    }
  ]
}
