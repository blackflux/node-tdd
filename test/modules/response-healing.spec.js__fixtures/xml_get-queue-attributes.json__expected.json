[
  {
    "scope": "https://sqs.us-west-2.amazonaws.com:443",
    "method": "POST",
    "path": "/",
    "body": {
      "AttributeNames": [
        "ApproximateNumberOfMessages",
        "ApproximateNumberOfMessagesDelayed",
        "ApproximateNumberOfMessagesNotVisible"
      ],
      "QueueUrl": "some-queue-url"
    },
    "status": 200,
    "response": {
      "Attributes": {
        "QueueArn": "arn:aws:sqs:us-west-2:012345678910:some-queue-arn",
        "ApproximateNumberOfMessagesNotVisible": "0",
        "ApproximateNumberOfMessagesDelayed": "0",
        "ApproximateNumberOfMessages": "1"
      }
    },
    "responseIsBinary": false,
    "reqheaders": {
      "amz-sdk-invocation-id": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
      "amz-sdk-request": "attempt=1; max=3",
      "authorization": "^AWS4-HMAC-SHA256 Credential=.+$",
      "content-length": "^[1-9][0-9]*$",
      "content-type": "application/x-amz-json-1.0",
      "host": "sqs.us-west-2.amazonaws.com",
      "user-agent": "^aws-sdk-js/.+$",
      "x-amz-content-sha256": "^[a-f0-9]{64}$",
      "x-amz-date": "^[0-9]{8}T[0-9]{6}Z$",
      "x-amz-target": "AmazonSQS.GetQueueAttributes",
      "x-amz-user-agent": "^aws-sdk-js/[0-9]+.[0-9]+.[0-9]+$",
      "x-amzn-query-mode": "true"
    }
  }
]
